src/marshmallow_sqlalchemy/convert.py

```diff
@@ -35,6 +39,32 @@ def _postgres_array_factory(converter, data_type):
     )
 
 
+def _set_meta_kwarg(field_kwargs, key, value):
+    if _META_KWARGS_DEPRECIATED:
+        field_kwargs["metadata"][key] = value
+    else:
+        field_kwargs[key] = value
+
+
+def _field_update_kwargs(field_class, field_kwargs, kwargs):
+    if not kwargs:
+        return field_kwargs
+
+    possible_field_keywords = {
+        key
+        for cls in inspect.getmro(field_class)
+        for key, param in inspect.signature(cls).parameters.items()
+        if param.kind is inspect.Parameter.POSITIONAL_OR_KEYWORD
+        or param.kind is inspect.Parameter.KEYWORD_ONLY
+    }
+    for k, v in kwargs.items():
+        if k in possible_field_keywords:
+            field_kwargs[k] = v
+        else:
+            _set_meta_kwarg(field_kwargs, k, v)
+    return field_kwargs
+
+
```

Comment:
Determine the field's possible keyword arguments and explicitly set arguments that aren't a part of that set as metadata.